### PR TITLE
Extracted header changes fror AP_Custom

### PR DESCRIPTION
**Pull request by peterbarker**
In preparation for AP_Custom (or whatever we call it...)

Randy pointed out we should have a noun, and we had these suggestions:

`AP_CustomVehicle`?
`AP_GenericVehicle`?

I think I like `AP_CustomVehicle` as someone pointed out it wouldn't be a bad target for the custom build server...

```
Board               AP_Periph  antennatracker  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *               *      *           *       *                 *      *      *
HerePro             0                                                                                     
Hitec-Airspeed      *                                                                                     
KakuteH7-bdshot                *               *      *           *       *                 *      *      *
MatekF405                      *               *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot             *               *                  *       *                 *      *      *
f103-QiotekPeriph   *                                                                                     
f303-Universal      *                                                                                     
iomcu                                                                           *                         
revo-mini                      *               *      *           *       *                 *      *      *
skyviper-v2450                                                    *                                       
```

### CODE REVIEW

**Comment by rmackay9** [comment]
I'd vote for "Vehicle", "GenericVehicle", "EmptyVehicle" or "NullVehicle".  I'm not a huge fan of "CustomVehicle" because nearly all AP vehicles are customised.

File: libraries/AP_Vehicle/AP_Vehicle.cpp [hunk 1]
```diff
@@ -682,10 +686,13 @@ void AP_Vehicle::send_watchdog_reset_statustext()
 
 bool AP_Vehicle::is_crashed() const
 {
+#if AP_ARMING_ENABLED
     if (AP::arming().is_armed()) {
         return false;
     }
     return AP::arming().last_disarm_method() == AP_Arming::Method::CRASH;
+#endif
```
Comment (tridge) [inline libraries/AP_Vehicle/AP_Vehicle.cpp:694]: #else

**Comment by tridge** [comment]
good to merge with minor #else change

File: libraries/AP_Vehicle/AP_Vehicle.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Vehicle/AP_Vehicle.cpp]: Done

**Pull request merged**